plugins/node/opentelemetry-instrumentation-mongodb/test/mongodb-v4-v5.metrics.test.ts

```diff
@@ -28,7 +28,27 @@ import {
   ResourceMetrics,
 } from '@opentelemetry/sdk-metrics';
 
-import * as mongodb from 'mongodb';
+import {
+  getInstrumentation,
+  registerInstrumentationTesting,
+} from '@opentelemetry/contrib-test-utils';
+
+import * as assert from 'assert';
+
+// Get instrumentation (singleton)
+let instrumentation: MongoDBInstrumentation;
+{
+  const instance: MongoDBInstrumentation | undefined = getInstrumentation();
+  if (!instance) {
+    instrumentation = new MongoDBInstrumentation();
+    registerInstrumentationTesting(instrumentation);
+  } else {
+    instrumentation = instance;
+  }
+}
```

Comment:
Hmm, what's the reason to do it like that vs. what how it was before? Was `registerInstrumentationTesting()` not working as intended? :thinking:

Reply:
Following your comment, I have adapted to make it less complex, but in short I had some issues with the singleton and the fact that re-calling [registerInstrumentationTesting](https://github.com/open-telemetry/opentelemetry-js-contrib/blob/main/packages/opentelemetry-test-utils/src/instrumentations/instrumentation-singelton.ts#L37) disables the new instrumentation and returns the old one.

It works fine in the tests now because

- `test/mongodb-v4-v5.metrics.test.ts` creates a new instrumentation and registers it - it never disables it
- `test/mongodb-v4.test.ts` creates a new instrumentation and tries to register it
  - But actually the new instrumentation is enabled (constructor) and then directly disabled by registerInstrumentationTesting. The old one, still active, is returned
- `test/mongodb-v4.test.ts` disables the instrumentation at the end of the tests

I only noticed that recently, that the order of test files is important :/
The other way around doesn't work because the traces' tests disable the instrumentation at the end, making the metrics tests run/fail without any instrumentation.